images/inference-services/ovms/run.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+
+set -ex
+set -u
+set -o pipefail
+
+# load org and project from repository if exists,
+# if not, set them as a random string
+if [ -e ".fuseml/_project" ]; then
+    ORG=$(cat .fuseml/_org)
+    PROJECT=$(cat .fuseml/_project)
+else
+    ORG=$(head /dev/urandom | LC_ALL=C tr -dc a-z0-9 | head -c 6)
+    PROJECT=$(head /dev/urandom | LC_ALL=C tr -dc a-z0-9 | head -c 6)
+fi
+
+NAMESPACE=${FUSEML_ENV_WORKFLOW_NAMESPACE}
+OVMS_ISTIO_GATEWAY=ovms-gateway
+
+# Extracting the domain name is done by looking at the Istio Gateway created in the
+# same namespace by the OVMS installer extension and extracting the domain out of the
+# host wildcard configured in the form of '*.<domain>'
+DOMAIN=$(kubectl get Gateway ${OVMS_ISTIO_GATEWAY} -n ${NAMESPACE} -o jsonpath='{.spec.servers[0].hosts[0]}')
+ISTIO_HOST="${ORG}.${PROJECT}${DOMAIN/\*/}"
+APP_NAME=${ORG}-${PROJECT}
+
+cat << EOF > /opt/openvino/templates/values.yaml
+#@data/values
+---
+namespace: "${NAMESPACE}"
+name_suffix: "${APP_NAME}"
+labels:
+  fuseml/app-name: "${PROJECT}"
+  fuseml/org: "${ORG}"
+  fuseml/app-guid: "${ORG}.${PROJECT}"
+  fuseml/workflow: "${FUSEML_ENV_WORKFLOW_NAME}"
+ovms_image_tag: "${FUSEML_OVMS_IMAGE_TAG}"
+aws_access_key_id: "${AWS_ACCESS_KEY_ID}"
+aws_secret_access_key: "${AWS_SECRET_ACCESS_KEY}"
+s3_compat_api_endpoint: "${S3_ENDPOINT}"
+model_path: "${FUSEML_MODEL}"
+model_name: "${FUSEML_MODEL_NAME}"
+istio_host: "${ISTIO_HOST}"
+istio_gateway: "${OVMS_ISTIO_GATEWAY}"
+EOF
+
+ytt -f /opt/openvino/templates/ | kubectl apply -f -
+
+# rollout fails if the object does not exist yet, so we need to wait until it is created
+count=0
+until kubectl get deploy ovms-${APP_NAME} -n ${FUSEML_ENV_WORKFLOW_NAMESPACE} 2>/dev/null; do
+  count=$((count + 1))
+  if [[ ${count} -eq "30" ]]; then
+    echo "Timed out waiting for Deployment to exist"
+    exit 1
+  fi
+  sleep 2
+done
+
+#TODO: if the deployment has already failed during a previous run, this will fail
+kubectl rollout status deploy/ovms-${APP_NAME} -n ${FUSEML_ENV_WORKFLOW_NAMESPACE}
```

Comment:
Maybe you could move this to a function into the base image, and seldon could also make use of it?